solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -857,31 +851,33 @@ public int size() {
    * </ul>
    * The terms are tokenized with the query analyzer.
    */
-  protected class SubsetMatchElevationProvider implements ElevationProvider {
+  protected class DefaultElevationProvider implements ElevationProvider {
 
-    private final SubsetMatcher<String, Elevation> subsetMatcher;
+    private final TrieSubsetMatcher<String, Elevation> subsetMatcher;
     private final Map<String, Elevation> exactMatchElevationMap;
 
     /**
-     * @param subsetMatcherBuilder The {@link SubsetMatcher.Builder} to build the {@link SubsetMatcher}.
+     * @param subsetMatcherBuilder The {@link TrieSubsetMatcher.Builder} to build the {@link TrieSubsetMatcher}.
      * @param elevationBuilderMap The map of elevation rules.
      */
-    protected SubsetMatchElevationProvider(SubsetMatcher.Builder<String, Elevation> subsetMatcherBuilder,
-                                           Map<ElevatingQuery, ElevationBuilder> elevationBuilderMap) {
+    protected DefaultElevationProvider(TrieSubsetMatcher.Builder<String, Elevation> subsetMatcherBuilder,
+                                       Map<ElevatingQuery, ElevationBuilder> elevationBuilderMap) {
       exactMatchElevationMap = new LinkedHashMap<>();
       Collection<String> queryTerms = new ArrayList<>();
-      StringBuilder concatenatedTerms = new StringBuilder();
+      Consumer<String> termsConsumer = queryTerms::add;
```

Comment:
Is pulling the method reference outside the loop actually more efficient?  Please point to info showing this.  I'm kinda sad if it is... I like the clean/simplicity of it used where needed.  @uschindler I have a feeling you know

Reply:
Yes it makes a difference. The lambda instance is created once, compared to inside the loop where a lambda instance is created for each step of the loop because the collection is not static.